routers/web/explore/code.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package explore
+
+import (
+	"net/http"
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	code_indexer "code.gitea.io/gitea/modules/indexer/code"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+const (
+	// tplExploreCode explore code page template
+	tplExploreCode base.TplName = "explore/code"
+)
+
+// Codes render explore code page
+func Codes(ctx *context.Context) {
```

Comment:
Not really liking this name... I think there's some slight inconsistency with the naming of functions in this package.

Reply:
The name was changed from `ExploreCode` to `Codes` which like `ExploreRepos` to `Repos`.

Reply:
I guess `Code` is already a plural - so I think the `s` is triggering me. Would be happier if you dropped the `s` I think.

```suggestion
func Code(ctx *context.Context) {
```

---

But I think there is still a structural issue with the names of page renderers and their sub components here (and elsewhere).

Within this package there are the `Render*` functions which admittedly have a different signature but everything else in this package is also rendering. So I'm not sure that they need the `Render` prefix, but I do see the point of having some distinct way of labelling things as incomplete pages. Maybe these should be suffixed `Component` as they're not full pages. e.g. `UserSearchComponent` (or moved out of the `explore` package so the only things it exports are actually full pages.)

If we went for the suffix approach, that would then mean that the full page renderers here & elsewhere should really gain a `Page` suffix - e.g. `CodePage`, `UsersPage`, `OrganizationsPage` and so on. However, I appreciate that that would make the web routes function a bit wordy - so maybe cleaning up the routers packages so that only full page rendering functions are ever exported from them with subpage components elsewhere might be a better idea.

With better and clearer labelling of full page rendering functions etc we'll be better placed to actually start documenting the renderers as to which paths it's expected to fire on, it's permissions, etc. and maybe even move to make routes functions be generated.

Reply:
I will rename it to `Code` but not with `Page` suffix, otherwise all route functions will be added `Page`. It's awful.

Reply:
It's not great and I understand the reluctance - however there is a genuine issue here. Here are a couple of examples the problem that adding Page suffices or cleaning up the packages would help with:

in `routers/repo/attachment.go:94` there is a func called `GetAttachment(ctx context.Context)` is this a page renderer? user.GetUserParams isn't a renderer. Turns out that it is a renderer.

RefBlame in routers/repo/blame.go is a page renderer, as is RefCommits in routers/repo/commits.go, Commits in that file is not a direct page renderer though nor are the RepoRefByType functions.

Would you guess that InitializeLabels is actually a page renderer but that RetrieveLabels is not?

From looking at a signature and name of a function you can't tell what it's supposed to be doing - and the comments aren't much better.

Reply:
should we make the diff even more complex?

I think it's a valid issue but does it meet the scope of this pull?